packages/@divvi/mobile/src/components/TokenDisplayV2.tsx

```diff
@@ -0,0 +1,167 @@
+   */
+  style?: StyleProp<TextStyle>
+
+  /**
+   * Optional sign to indicate the type of token amount display.
+   *   - `none`: shows number without the sign
+   *   - `only-negative` adds `-` sign in front of the negative amount, positive amount is ignored
+   *   - `show-all` adds whether `+` or `-` based on the amount
+   */
+  sign?: 'none' | 'only-negative' | 'show-all'
+
+  /**
+   * ID of the token to display. If not provided – error will be thrown.
+   */
+  tokenId: string | undefined
+
+  /**
+   * The amount of the token to display.
+   */
+  tokenAmount: BigNumber.Value
+}
+
+/**
+ * @returns
+ * - if token cannot be found by `tokenId` then returns `null` and logs an error
+ * - otherwise returns value in the format: `1.123456 USDC`
+ */
+export function TokenAmountDisplay(props: Props) {
+  const tokenInfo = useTokenInfo(props.tokenId)
+  const tokenAmount = new BigNumber(props.tokenAmount)
+  const sign = getSign(props.sign, tokenAmount)
+
+  if (!tokenInfo) {
+    Logger.error(TAG, 'tokenInfo not found')
+    return null
+  }
+
+  return (
+    <Text testID={props.testID} style={props.style}>
+      <>
+        {props.approx && `${APPROX_SYMBOL} `}
+        {sign}
+        {getTokenAmountDisplay(tokenAmount, tokenInfo)}
+      </>
+    </Text>
+  )
+}
+
+/**
+ * @returns Returns value in the following formats:
+ * - `Price Unavailable` when local amount cannot be determined
+ * - `$1.23`
+ */
+export function LocalAmountDisplay(props: Props) {
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol) ?? LocalCurrencySymbol.USD
+  const tokenInfo = useTokenInfo(props.tokenId)
+  const tokenAmount = new BigNumber(props.tokenAmount)
+  const localAmount = useTokenToLocalAmount(tokenAmount, tokenInfo?.tokenId)
+  const sign = getSign(props.sign, tokenAmount)
+
+  return (
+    <Text testID={props.testID} style={props.style}>
+      {localAmount ? (
+        <>
+          {props.approx && `${APPROX_SYMBOL} `}
+          {sign}
+          {getLocalAmountDisplay(localAmount, localCurrencySymbol)}
+        </>
+      ) : (
+        <Trans i18nKey="localAmount_noFiatPrice" />
+      )}
+    </Text>
+  )
+}
+
+/**
+ * @returns
+ * - if token cannot be found by `tokenId` then returns `null` and logs an error
+ * - otherwise returns value in the following formats:
+ *   - `1.123456 USDC ($1.23)` - when local amount is available
+ *   - `1.123456 USDC` - when local amount is not available
+ */
+export function TokenAndLocalAmountDisplay(props: Props) {
+  const tokenInfo = useTokenInfo(props.tokenId)
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol) ?? LocalCurrencySymbol.USD
+  const tokenAmount = new BigNumber(props.tokenAmount)
+  const localAmount = useTokenToLocalAmount(tokenAmount, tokenInfo?.tokenId)
+  const localAmountDisplay = getLocalAmountDisplay(localAmount, localCurrencySymbol)
+  const sign = getSign(props.sign, tokenAmount)
+
+  if (!tokenInfo) {
+    Logger.error(TAG, 'tokenInfo not found')
```

Comment:
Maybe skip those, usually very noisy if during render. Or maybe only keep in __DEV__?